wizard/state.py

```diff
@@ -53,3 +62,75 @@ class Metadata:
 
     def __bool__(self):
         return self.name != "" and self.url != ""
+    
+@dataclasses.dataclass
+class Croissant:
+    metadata: Metadata = Metadata()
+    distributions: list[Distribution] = dataclasses.field(default_factory=list)
+    record_sets: list[RecordSet] = dataclasses.field(default_factory=list)
+    def update_metadata(self, metadata: Metadata) -> None:
+        self.metadata = Metadata
+    def add_distribution(self, distribution: Distribution) -> None:
+        self.distributions.append(distribution)
+    def update_distribution(self, key: int, distribution: Distribution) -> None:
+        self.distributions[key] = distribution
+    def remove_distribution(self, key: int) -> None:
+        del self.distributions[key]
+    def add_record_set(self, record_set: RecordSet) -> None:
+        self.record_sets.append(record_set)
+    def update_record_set(self, key: int, record_set: RecordSet) -> None:
+        self.record_sets[key] = record_set
+    def remove_record_set(self, key: int) -> None:
+        del self.record_sets[key]
+
+def CanonicalToWizard(dataset: mlc.Dataset) -> Croissant:
```

Comment:
This could be a [`classmethod`](https://docs.python.org/3/library/functions.html#classmethod) of `Croissant` and be called `from_canonical`.

Like this you'd write:

```python
croissant = Croissant.from_canonical(dataset)
```